libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs

```diff
@@ -346,6 +359,49 @@ public QnAMakerDialog([CallerFilePath] string sourceFilePath = "", [CallerLineNu
         [JsonProperty("rankerType")]
         public StringExpression RankerType { get; set; } = new StringExpression(RankerTypes.DefaultRankerType);
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to include precise answer in response. 
+        /// </summary>
+        /// <value>
+        /// True or false to include precise answer in response, defaults to false. 
+        /// </value>
+        [JsonProperty("enablePreciseAnswer")]
+        public bool EnablePreciseAnswer { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to include unstructured sources in search for answers. 
+        /// </summary>
+        /// <value>
+        /// True/False indicating whether to include unstructured sources, defaults to false. 
+        /// </value>
+        [JsonProperty("includeUnstructuredSources")]
+        public bool IncludeUnstructuredSources { get; set; }
+
+        /// <summary>
+        /// Gets or sets the metadata and sources used to filter QnA Maker results.
+        /// </summary>
+        /// <value>
+        /// An object with metadata, source filters and corresponding operators.
+        /// </value>
+        [JsonProperty("filters")]
+        public ObjectExpression<Filters> Filters { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the dialog response should display only precise answers.
+        /// </summary>
+        /// <value>
+        /// True/False, defaults to False.
+        /// </value>
+        [JsonProperty("displayPreciseAnswerOnly")]
+        public bool DisplayPreciseAnswerOnly { get; set; }
```

Comment:
Use BoolExpression here instead of native type to enable future integration with composer if needed